Trappist/src/Promact.Trappist.Core/Controllers/TestController.cs

```diff
@@ -99,13 +101,15 @@ public TestsController(ITestsRepository testRepository)
         /// <param name="testObject">The parameter "testObject" is an object of Test</param>
         /// <returns>Updated Settings of that Test</returns>     
         [HttpPut("{id}/settings")]
-        public async Task<IActionResult> UpdateTestSettings([FromRoute] int id, [FromBody] Test testObject)
+        public async Task<IActionResult> UpdateTestByIdAsync([FromRoute] int id, [FromBody] Test testObject)
         {
+            if (!await _testRepository.IsTestNameUniqueAsync(testObject.TestName,id))
+                return BadRequest();
```

Comment:
please check bad request for testobject also. Please add model error to identify at client end.